startup/27-vstream-cam.py

```diff
@@ -105,30 +122,29 @@ def trigger(self, *args, **kwargs):
         # Averaging over all frames and summing 3 RGB channels
         averaged = frames.mean(axis=0).sum(axis=-1)
 
-        with h5py.File(data_file, "w") as f:
-            group = f.create_group("/entry")
-            group.create_dataset("frames", data=frames, compression="lzf")
-            group.create_dataset("averaged", data=averaged, compression="lzf")
+        current_frame = next(self._counter)
+        self._dataset.resize((current_frame + 1, *self._frame_shape))
```

Comment:
It is worth checking the performance of resizing every time and resizing in steps of the chunk size.  If you set the fill value and we record enough meta-data to slice to just the frame we want it is not the end of the world if there are nan frames at the end. 

You can also truncate in unstage.